plugin/trino-exasol/src/main/java/io/trino/plugin/exasol/ExasolClientModule.java

```diff
@@ -0,0 +1,70 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.exasol;
+
+import com.exasol.jdbc.EXADriver;
+import com.google.inject.Binder;
+import com.google.inject.Provides;
+import com.google.inject.Scopes;
+import com.google.inject.Singleton;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
+import io.opentelemetry.api.OpenTelemetry;
+import io.trino.plugin.jdbc.BaseJdbcConfig;
+import io.trino.plugin.jdbc.ConfiguringConnectionFactory;
+import io.trino.plugin.jdbc.ConnectionFactory;
+import io.trino.plugin.jdbc.DriverConnectionFactory;
+import io.trino.plugin.jdbc.ForBaseJdbc;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcStatisticsConfig;
+import io.trino.plugin.jdbc.credential.CredentialProvider;
+import io.trino.plugin.jdbc.ptf.Query;
+import io.trino.spi.function.table.ConnectorTableFunction;
+
+import java.util.Properties;
+
+import static com.google.inject.multibindings.Multibinder.newSetBinder;
+import static io.airlift.configuration.ConfigBinder.configBinder;
+
+public class ExasolClientModule
+        extends AbstractConfigurationAwareModule
+{
+    @Override
+    protected void setup(Binder binder)
+    {
+        binder.bind(JdbcClient.class).annotatedWith(ForBaseJdbc.class).to(ExasolClient.class).in(Scopes.SINGLETON);
+        configBinder(binder).bindConfig(JdbcStatisticsConfig.class);
+        newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
+    }
+
+    @Provides
+    @Singleton
+    @ForBaseJdbc
+    public static ConnectionFactory connectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider, OpenTelemetry openTelemetry)
+    {
+        Properties connectionProperties = new Properties();
+        return new ConfiguringConnectionFactory(
+                DriverConnectionFactory.builder(
+                                new EXADriver(),
+                                config.getConnectionUrl(),
+                                credentialProvider)
+                        .setConnectionProperties(connectionProperties)
+                        .setOpenTelemetry(openTelemetry)
+                        .build(),
+                connection -> {
+                    // Deactivate SNAPSHOT_MODE (https://docs.exasol.com/db/latest/database_concepts/snapshot_mode.htm)
+                    // to ensure that {@link Connection#getMetaData()} always returns up-to-date data.
+                    connection.createStatement().execute("ALTER SESSION SET SNAPSHOT_MODE = 'OFF'");
```

Comment:
is `snapshottransactions` JDBC connection property same or different?

It's better to configure things on connection to avoid sending an additional query for every new connection.